### PR TITLE
feat: add `haversineDistance`

### PR DESCRIPTION
**Pull request by vil02**
This PR adds a `haversineDistance` function.

I realize that the `close` function is _ugly_. I could not force the [`almostEqual`](https://nim-lang.org/docs/math.html#almostEqual%2CT%2CT%2CNatural) to work here.

### CODE REVIEW

File: maths/haversine_distance.nim [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+import std/[math]
+
+func haversineDistance(latitudeA, longitudeA, latitudeB,
+    longitudeB: float): float =
+  ## returns the length of the shortest path connecting the input points on an unit sphere.
+  ## The input points are represented by their spherical/geographical coordinates.
+  let dLatitude = latitudeB - latitudeA
+  let dLongitude = longitudeB - longitudeA
+  let a = sin(dLatitude / 2.0)^2+cos(latitudeA)*cos(latitudeB)*sin(dLongitude / 2.0)^2
```
Comment (vil02) [inline maths/haversine_distance.nim:11]: Funny thing: if I:
```suggestion
  let a = sin(dLatitude /2.0)^2+cos(latitudeA)*cos(latitudeB)*sin(dLongitude / 2.0)^2
```
I get:
```
.../Nim/maths/haversine_distance.nim(11, 25) Error: type mismatch
Expression: /2.0
  [1] 2.0: float64

Expected one of (first mismatch at [position]):
[1] proc `/`(x, y: int): float
[2] proc `/`(x, y: float): float
[2] proc `/`(x, y: float32): float32
```

I am using
```
Nim Compiler Version 2.0.0 [Linux: amd64]
Compiled at 2023-08-01
Copyright (c) 2006-2023 by Andreas Rumpf

git hash: a488067a4130f029000be4550a0fb1b39e0e9e7c
active boot switches: -d:release
```

Comment (ZoomRmc) [reply maths/haversine_distance.nim:12]: What's funny about it?
```nim
let (a, b) = (1.0, 2.0)
echo a / b # 0.5
echo a /b  # parsed as ```echo(a, b.`/`)```
```

Comment (vil02) [reply maths/haversine_distance.nim:12]: You are right. It is not funny at all.

Comment (ZoomRmc) [reply maths/haversine_distance.nim:12]: Now we're talking! :)

File: maths/haversine_distance.nim [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+      (-PI / 2.0, 0.0, -PI / 4.0, 0.0, PI / 4.0),
+      (-PI / 2.0, 0.0, -PI / 4.0, 0.6, PI / 4.0),
+      (-PI / 2.0, 3.0, -PI / 4.0, 0.2, PI / 4.0),
+      ].mapIt:
+        (id: fmt"posA=({it[0]}, {it[1]}), posB=({it[2]}, {it[3]})",
+         latitudeA: it[0], longitudeA: it[1],
+         latitudeB: it[2], longitudeB: it[3],
+         expected: it[4])
+
+    func close(a, b: float): bool =
```
Comment (ZoomRmc) [inline maths/haversine_distance.nim:42]:
```suggestion
    func isClose(a, b: float): bool =
```

Due to extensive use of overloading in Nim, `isAdjective(): bool` and `verb()` are idiomatic, especially when adjective==verb.

Comment (vil02) [reply maths/haversine_distance.nim]: Done in 470d5bbce7e4a67448006f63cbc3deac9b4e2fcb.

File: maths/haversine_distance.nim [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+func haversineDistance(latitudeA, longitudeA, latitudeB,
+    longitudeB: float): float =
+  ## returns the length of the shortest path connecting the input points on an unit sphere.
+  ## The input points are represented by their spherical/geographical coordinates.
+  ## The inputs are expected to be in radians.
+  let dLatitude = latitudeB - latitudeA
+  let dLongitude = longitudeB - longitudeA
+  let a = sin(dLatitude / 2.0)^2+cos(latitudeA)*cos(latitudeB)*sin(dLongitude / 2.0)^2
+  return 2.0*arcsin(sqrt(a))
```
[the remark this answers is not in the document]
Comment (vil02) [reply maths/haversine_distance.nim]: Thanks for a reminder - I keep forgetting about that. It would be great to have some automated check for that.

**Review by ZoomRmc — APPROVED** [review]

**Review by dlesnoff — APPROVED** [review]

**Pull request merged**